README.md

```diff
@@ -7,40 +7,173 @@
+
+<!-- START doctoc generated TOC please keep comment here to allow auto update -->
+<!-- DON'T EDIT THIS SECTION, INSTEAD RE-RUN doctoc TO UPDATE -->
+**Table of Contents**
+
+- [Overview of contribution flow](#overview-of-contribution-flow)
+- [Glossary](#glossary)
+- [Contribution stage A. Setup the fork and local clone](#contribution-stage-a-setup-the-fork-and-local-clone)
+- [Contribution stage B. Adding new code and updating submissions](#contribution-stage-b-adding-new-code-and-updating-submissions)
+  - [B1. Submitting new task (app) code base](#b1-submitting-new-task-app-code-base)
+  - [B2. When any changes are required](#b2-when-any-changes-are-required)
+  - [B3. When all of your PRs are finally merged](#b3-when-all-of-your-prs-are-finally-merged)
+- [Reference materials and troubleshooting](#reference-materials-and-troubleshooting)
+
+<!-- END doctoc generated TOC please keep comment here to allow auto update -->
+<!-- *generated with [DocToc](https://github.com/thlorenz/doctoc)* -->
+
+## Overview of contribution flow
+
+Now you are a contributor to an educational open source project.
+
+[These slides](https://docs.google.com/presentation/d/13dati5gvA5f_hQFgxJPhPicjF5CRKu1e75RSsahmEaU/edit#slide=id.g58afea5148_0_0)
+visualize the contribution workflow.
+
+The instructions below will lead you through the contribution workflow
+as you will submit your tasks code base.
+
+## Glossary
+
+Term | Definition
+---  | ---
+repo                   | short for 'repository'
+_homeworks main repo_  | Kottans FE course homeworks repository: https://github.com/kottans/frontend-2021-homeworks
+_homeworks fork_       | your own fork of homeworks main repo
+_homeworks local repo_ | your local clone of your homeworks fork
+app                    | an application (or web page) you develop to fulfil a specific task in the course
+_app repo_             | a repository that contains your application code
+
+## Contribution stage A. Setup the fork and local clone
+
+You need to do this only once.
+
+A1. Fork _homeworks main repo_ via GitHub web interface
+
+A2. Clone _homeworks fork_ on your local machine:
+   - `git clone https://github.com/YOUR_USERNAME/frontend-2021-homeworks.git`
+
+> Make sure you are inside the homeworks local repo directory before any further operations  
+
+A3. Add _homeworks main repo_ as an upstream:
+   - `git remote add upstream https://github.com/kottans/frontend-2021-homeworks.git`
+
+A4. In _homeworks local repo_, add a folder with your github username under `submissions` directory
+
+You will add code from your every task (app) in its individual directory under
+your named directory. So, the file structure will be similar to the following
+(in this example `amashoshyna` is a username and `js-dom` is a task name)
+
+![File structure example](img/file-structure.png)
+
+## Contribution stage B. Adding new code and updating submissions 
+
+### B1. Submitting new task (app) code base
+
```

Comment:
```suggestion

> IMPORTANT things to remember at all times.
>
> 1. Do not commit to `main`.
>
> 2. `checkout main` before creating a new branch, i.e. do not create new branch while
> on any branch that is not `main`.
>
> 3. Do not merge your branches into `main`. You can merge `main` into your branch if you
> realize why you need this. In current workflow you normally do not need this.
>
> And don't panic if some code seems to be gone when you switch between branches.
> It will be available to you once you switch back to a specific branch.
> The point is to submit tasks in isolation from each other.

```